lib/identity/stateTransitions/identityCreateTransition/validateLockTransactionFactory.js

```diff
@@ -34,10 +64,38 @@ function validateLockTransactionFactory(getLockedTransactionOutput) {
       }
     }
 
+    if (!rawTransaction) {
+      result.addError(new IdentityLockTransactionNotFoundError(transactionHash));
+    }
+
     if (!result.isValid()) {
       return result;
     }
 
+    const transaction = new Transaction(rawTransaction.hex);
+
+    if (!transaction.outputs[outputIndex]) {
+      result.addError(new InvalidIdentityOutPointError(`Output with index ${outputIndex} not found`));
+
+      return result;
+    }
+
+    if (useLockTxFallback) {
```

Comment:
Agreed to move this logic to `getLockedTransactionOutput` and rename this function to `fetchLockTransactionOutput`

Reply:
Do you mean to move lock validation to that function?

Reply:
Yeah, part of it is there already.

Reply:
And do you want it to throw an error, that we can catch here and add to result?

Reply:
I'm not sure what benefits we can have from it, why just don't keep it here?

Reply:
We are already checking that transaction is not found in `fetchLockTransactionOutPoint` and throw an exception if not so we could add one more check that transaction is not confirmed.

We can't just keep it here 'coz `fetchLockTransactionOutPoint` returns outpoint, not transaction so we have to duplicate a lot of code.

Reply:
Yeah, I know, my point was that why just don't keep it here? `fetch` doesn't represent the fact that we already validated some stuff in there, is it?

Reply:
We were using this function initially in two places: in the validation function and apply ST. We could remove validation logic from `fetch` method but we need to avoid code duplication.

Reply:
Agreed to rename the function.